blueocean-bitbucket-pipeline/src/test/java/io/jenkins/blueocean/blueocean_bitbucket_pipeline/server/BitbucketServerEndpointTest.java

```diff
@@ -0,0 +1,265 @@
+        validBitbucketServer(true);
+        Map resp = request()
+                .status(400)
+                .jwtToken(token)
+                .data(ImmutableMap.of())
+                .post(URL)
+                .build(Map.class);
+        Assert.assertNotNull(resp);
+
+        List errors = (List) resp.get("errors");
+        assertEquals(2, errors.size());
+
+        Map error1 = (Map) errors.get(0);
+        assertEquals("name", error1.get("field"));
+
+        Map error2 = (Map) errors.get(1);
+        assertEquals("apiUrl", error2.get("field"));
+    }
+
+    @Test
+    public void testMissingUrlParam() throws Exception {
+        validBitbucketServer(true);
+        Map resp = request()
+                .status(400)
+                .jwtToken(token)
+                .data(ImmutableMap.of("name", "foo"))
+                .post(URL)
+                .build(Map.class);
+        Assert.assertNotNull(resp);
+
+        List errors = (List) resp.get("errors");
+        assertEquals(1, errors.size());
+
+        Map error1 = (Map) errors.get(0);
+        assertEquals("apiUrl", error1.get("field"));
+        assertEquals("MISSING", error1.get("code"));
+        assertNotNull(error1.get("message"));
+    }
+
+    @Test
+    public void testMissingNameParam() throws Exception {
+        validBitbucketServer(true);
+        Map resp = request()
+                .status(400)
+                .jwtToken(token)
+                .data(ImmutableMap.of("apiUrl", getApiUrl()))
+                .post(URL)
+                .build(Map.class);
+        Assert.assertNotNull(resp);
+
+        List errors = (List) resp.get("errors");
+        assertEquals(1, errors.size());
+
+        Map error1 = (Map) errors.get(0);
+        assertEquals("name", error1.get("field"));
+        assertEquals("MISSING", error1.get("code"));
+        assertNotNull(error1.get("message"));
+    }
+
+    @Test
+    public void avoidDuplicateByUrl() throws Exception {
+        validBitbucketServer(true);
+        // Create a server
+        Map server = request()
+                .status(200)
+                .jwtToken(token)
+                .data(ImmutableMap.of(
+                        "name", "My Server",
+                        "apiUrl", getApiUrl()
+                ))
+                .post(URL)
+                .build(Map.class);
+
+        // Create a server
+        Map resp = server = request()
+                .status(400)
+                .jwtToken(token)
+                .data(ImmutableMap.of(
+                        "name", "My Server 2",
+                        "apiUrl", getApiUrl()
+                ))
+                .post(URL)
+                .build(Map.class);
+
+        List errors = (List) resp.get("errors");
+        assertEquals(1, errors.size());
+
+        Map error1 = (Map) errors.get(0);
+        assertEquals("apiUrl", error1.get("field"));
+        assertEquals("ALREADY_EXISTS", error1.get("code"));
+        assertNotNull(error1.get("message"));
```

Comment:
Might be good to add a test here with a trailing slash mismatch, just because it's very possibly two users may try to add the same server and omit a trailing slash in one but not the other.